proposals/dictionary-expressions.md

```diff
@@ -55,178 +55,86 @@ Choices here would have implications regarding potential syntactic ambiguities,
-
-### Question: Types that support both collection and dictionary initialization
-
-C# 12 supports collection types where the element type is some `KeyValuePair<,>`, where the type has an applicable `Add()` method that takes a single argument. Which approach should we use for initialization if the type also includes an indexer?
+Note: Many rules in this spec will refer to types needing to be the same `KeyValuePair<,>` type.  This is an informal way of saying the types must have an identity conversion between them.  As such, `KeyValuePair<(int X, int Y), object>` would be considered the same type a `KeyValuePair<(int, int), object?>` for the purpose of these rules.
 
-For example, consider a type like so:
+With a broad interpretation of these rules, all of the following would be legal:
 
 ```c#
-public class Hybrid<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
-{
-    public void Add(KeyValuePair<TKey, TValue> pair);
-    public TValue this[TKey key] { ... }
-}
+// Assigning to dictionary types:
+Dictionary<string, int> nameToAge1 = ["mads": 21, existingKvp];     // as would
+Dictionary<string, int> nameToAge2 = ["mads": 21, .. existingDict]; // as would
+Dictionary<string, int> nameToAge3 = ["mads": 21, .. existingListOfKVPS];
 
-// This would compile in C# 12:
-// Translating to calls to .Add.
-Hybrid<string, int> nameToAge = [someKvp];
+// Assigning to collection types:
+List<string, int> nameToAge1 = ["mads": 21, existingKvp];     // as would
+List<string, int> nameToAge2 = ["mads": 21, .. existingDict]; // as would
+List<string, int> nameToAge3 = ["mads": 21, .. existingListOfKVPS];
 ```
 
-Options include:
-
-1. Use applicable instance indexer if available; otherwise use C#12 initialization.
-2. Use applicable instance indexer if available; otherwise report an error during construction (or conversion?).
-3. Use C#12 initialization always.
-
-Resolution TBD.  Working group recommendation: Use applicable instance indexer only.  This ensures that everything dictionary-like is initialized in a consistent fashion.  This would be a break in behavior when recompiling.  The view is that these types would be rare.  And if they exist, it would be nonsensical for them to behave differently using the indexer versus the .Add (outside of potentially throwing behavior).
-
 ## Dictionary types
 
 A type is considered a *dictionary type* if the following hold:
 * The *element type* is `KeyValuePair<TKey, TValue>`.
 * The *type* has an instance *indexer* where:
   * The indexer has a single parameter with an identity conversion from the parameter type to `TKey`.\*
-  * There is an identity conversion from the indexer type to `TValue`.\*
+  * There is an identity conversion from the indexer return type to `TValue`.\*
```

Comment:
@cston  if this word is needed.  If the spec already talks about `indexer types` in other places, then we can revert.  I liked `return type` to make it clear which type this was referring to.

Reply:
`TValue` is only the return type from the getter, so I'd suggest leaving as "type" rather than "return type".

Reply:
done.